From 397a161eced5fad35879f595a18792006a3ca58e Mon Sep 17 00:00:00 2001
From: "Ruan, Hongfu" <hongfu.ruan@intel.com>
Date: Wed, 1 Dec 2021 14:37:52 +0800
Subject: [PATCH] change policy of REQUEST_INSTALL_PACKAGES to ALLOWED

this is required for 3rd-party app upgrade.

Tracked-On: OAM-100139
Signed-off-by: Ruan, Hongfu <hongfu.ruan@intel.com>
---
 core/java/android/app/AppOpsManager.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/java/android/app/AppOpsManager.java b/core/java/android/app/AppOpsManager.java
index 6baabb69e028..3872f1af1833 100644
--- a/core/java/android/app/AppOpsManager.java
+++ b/core/java/android/app/AppOpsManager.java
@@ -2305,7 +2305,7 @@ public class AppOpsManager {
             AppOpsManager.MODE_ALLOWED, // RUN_IN_BACKGROUND
             AppOpsManager.MODE_ALLOWED, // AUDIO_ACCESSIBILITY_VOLUME
             AppOpsManager.MODE_ALLOWED, // READ_PHONE_NUMBERS
-            AppOpsManager.MODE_DEFAULT, // REQUEST_INSTALL_PACKAGES
+            AppOpsManager.MODE_ALLOWED, // REQUEST_INSTALL_PACKAGES
             AppOpsManager.MODE_ALLOWED, // PICTURE_IN_PICTURE
             AppOpsManager.MODE_DEFAULT, // INSTANT_APP_START_FOREGROUND
             AppOpsManager.MODE_ALLOWED, // ANSWER_PHONE_CALLS
-- 
2.29.2

